### PR TITLE
Add to_collection method returning an Xcollection

### LINKED ISSUE #382: Use `Collection` object implementation from NCAR/xcollection in intake-esm
A few weeks ago, we experimented with [`Collection`](https://github.com/NCAR/xcollection), which provides a high-level mapping of name/key to Xarray.Datasets. The `xcollection.Collection` class is a dict-like class. It subclasses `collections.abc.MutableMapping` and provides additional functionality like
- `.map()`: applies a function to each dataset in the collection.
- `.choose()`: returns a collection with datasets containing all or any of the specified data variables.

Seems it would be convenient/neat to use this Collection in the `.to_dataset_dict()` method i.e. return a `Collection` object instead of a regular dictionary of datasets. Usage of this object is likely going to be experimental until https://github.com/TomNicholas/datatree is stable and ready for public usage.

### PR DESCRIPTION
**Pull request by jukent**
## Change Summary

Updating `to_dataset_dict` to return an xcollection.Collection object.

~~Needed to add `git+git://github.com/NCAR/xcollection.git` to requirements~~
And wrap dictionaries in `xc.Collection` 

There are some uses of the word "Collection" throughout the file that are not an `xc.Collection`, we might want to be very specific with our language throughout.

Tests seem to be unchanged by Collection vs Dictionary typing, maybe add an assert type test.

## Related issue number

- Fixes #382

## Checklist

- [x] Unit tests for the changes exist
- [x] Tests pass on CI
- [ ] Documentation reflects the changes where applicable

### CODE REVIEW

**Comment by jukent** [comment]
In the tests we have

```
)
def test_to_dataset_dict(path, query, xarray_open_kwargs):
    cat = intake.open_esm_datastore(path)
    cat_sub = cat.search(**query)
    _, ds = cat_sub.to_dataset_dict(xarray_open_kwargs=xarray_open_kwargs).popitem()
    assert 'member_id' in ds.dims
    assert len(ds.__dask_keys__()) > 0
    assert ds.time.encoding
```

But an xc.Collection won't have `dims`

**Comment by jukent** [comment]
@kmpaul This is failing with "Error: The process '/usr/share/miniconda3/condabin/mamba' failed with exit code 1" which seems to be the same as from the ProjectPythia pull requests (related to the new mamba version?). I'm looking for how you fixed that in those repositories now.

File: requirements.txt [hunk 1]
```diff
@@ -7,3 +7,4 @@ requests>=2.24.0
 xarray>=0.19
 zarr>=2.5
 pydantic>=1.8.2
+git+git://https://github.com/NCAR/xcollection.git
```
Comment (kmpaul) [inline requirements.txt:10]:
```suggestion
git+git://github.com/NCAR/xcollection.git
```

**Review by kmpaul — CHANGES_REQUESTED** [review]
Error in protocol

File: requirements.txt [hunk 2]
```diff
@@ -7,3 +7,4 @@ requests>=2.24.0
 xarray>=0.19
 zarr>=2.5
 pydantic>=1.8.2
+git+git://github.com/NCAR/xcollection.git
```
Comment (andersy005) [inline requirements.txt:10]: I think it needs to be a valid HTTP url

```suggestion
git+https://github.com/NCAR/xcollection
```

File: intake_esm/core.py [hunk 3]
```diff
@@ -441,9 +442,9 @@ def to_dataset_dict(
         aggregate: pydantic.StrictBool = None,
         skip_on_error: pydantic.StrictBool = False,
         **kwargs,
-    ) -> typing.Dict[str, xr.Dataset]:
+    ) -> typing.Collection[str, xc.Collection]:
```
Comment (andersy005) [inline intake_esm/core.py:445]:
```suggestion
    ) -> xc.Collection:
```

File: setup.py [hunk 4]
```diff
@@ -7,7 +7,7 @@
 from setuptools import find_packages, setup
 
 with open('requirements.txt') as f:
-    install_requires = f.read().strip().split('\n')
+    install_requires = f.read().strip().split('\n')[0:-1]
```
Comment (andersy005) [inline setup.py:10]: Is this change still needed?

File: intake_esm/core.py [hunk 5]
```diff
@@ -247,7 +248,7 @@ def __dir__(self) -> typing.List[str]:
         rv = [
             'df',
             'to_dataset_dict',
-            'to_dask',
+            'to_collection' 'to_dask',
```
Comment (andersy005) [inline intake_esm/core.py:251]:
```suggestion
            'to_collection',
            'to_dask',
```

**Review by andersy005 — APPROVED** [review]
Thank you for working on this, @jukent! Looks great!

**Review by kmpaul — APPROVED** [review]
Thanks, @jukent!

**Pull request merged**